### DISCUSSION TITLE
Error [ERR_MODULE_NOT_FOUND]

### DISCUSSION THREAD

**Discussion by matus123** [typedorm/typedorm#281] [Q&A]
hi,

i am trying to use https://github.com/typedorm library (https://github.com/typedorm/typedorm) with nextjs... but there is problem when i use this library in nextjs apis... i am getting runtime error
```
error - Error [ERR_MODULE_NOT_FOUND]: Cannot find module '/home/matus/repos/typedorm-nextjs-example/node_modules/@typedorm/core/esm/public-api' imported from /home/matus/repos/typedorm-nextjs-example/node_modules/@typedorm/core/esm/index.js
    at new NodeError (node:internal/errors:371:5)
    at finalizeResolution (node:internal/modules/esm/resolve:416:11)
    at moduleResolve (node:internal/modules/esm/resolve:932:10)
    at defaultResolve (node:internal/modules/esm/resolve:1044:11)
    at ESMLoader.resolve (node:internal/modules/esm/loader:422:30)
    at ESMLoader.getModuleJob (node:internal/modules/esm/loader:222:40)
    at ModuleWrap.<anonymous> (node:internal/modules/esm/module_job:76:40)
    at link (node:internal/modules/esm/module_job:75:36) {
  code: 'ERR_MODULE_NOT_FOUND'
}
```
i tried tinkering with tsconfig or webpack configs, but without any success

when i tried to use next-transpile-modules it started to work but getting these kind of warnings
```
warn  - ./node_modules/@typedorm/common/esm/src/helpers/load-package.js
Critical dependency: the request of a dependency is an expression

./node_modules/@typedorm/core/esm/src/classes/connection/connection-metadata-builder.js
Critical dependency: the request of a dependency is an expression
```
minimal repo: https://github.com/matus123/nextjs-typedorm-example
i opened discussion here aswell (https://github.com/vercel/next.js/discussions/40313)
i dont know if it is problem on nextjs or typedorm side

thanks

**Comment by whimzyLive** [comment]
I'll have a look. Meanwhile, can you try doing the same with the previous version 1.14.x?

**Reply by matus123** [reply]
it seems to work OK with 1.14.11 version

**Reply by whimzyLive** [reply]
@matus123 In 1.15.x we did some changes around how we publish modules for TypeDORM, to allow js bundlers to effectively bundle js code using TypeDORM, which I think is casing some issues here. 
I'll have a more in-depth look at it. Are you able to work with `1.14.x` until the fix is rolled out?

**Reply by matus123** [reply]
we are able to workaround this problem with `next-transpile-modules` and `1.15.x` version, but it is not ideal because we are getting those critical dependency warnings and are worried if it might break in future
